Coverage build fails with CMake 3.13 and later

Coverage build fails with CMake 3.13 and later, giving this error:

```
[ 43%] Linking C executable mysofa2json
cc: error: gcov: No such file or directory
make[2]: *** [src/CMakeFiles/mysofa2json.dir/build.make:102: src/mysofa2json] Error 1
make[2]: Leaving directory '/build/libmysofa-git/src/libmysofa/build'
make[1]: *** [CMakeFiles/Makefile2:131: src/CMakeFiles/mysofa2json.dir/all] Error 2
make[1]: Leaving directory '/build/libmysofa-git/src/libmysofa/build'
make: *** [Makefile:163: all] Error 2
```

`gcov` is on `PATH`:
```
$ command -v gcov
/usr/bin/gcov
```

This is caused by commit 6e4381b.

It looks like [target_link_options()](https://cmake.org/cmake/help/v3.16/command/target_link_options.html) accepts only link options (and not a command + link options), even when using the [INTERFACE](https://cmake.org/cmake/help/v3.16/prop_tgt/INTERFACE_LINK_OPTIONS.html#prop_tgt:INTERFACE_LINK_OPTIONS)  keyword. Being such, this change fixes the issue for me:

```diff
diff -Naurp a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2020-02-21 13:03:26.000000000 +0000
+++ b/src/CMakeLists.txt	2020-02-21 13:06:37.898161169 +0000
@@ -89,7 +89,7 @@ if(UNIX
   target_compile_options(mysofa-static PUBLIC -g -O0 -Wall -fprofile-arcs
                                               -ftest-coverage)
   if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
-    target_link_options(mysofa-static INTERFACE gcov --coverage)
+    target_link_options(mysofa-static INTERFACE --coverage)
   else()
     target_link_libraries(mysofa-static LINK_PUBLIC gcov --coverage)
   endif()
@@ -118,7 +118,7 @@ if(BUILD_SHARED_LIBS)
     target_compile_options(mysofa-shared PUBLIC -g -O0 -Wall -fprofile-arcs
                                                 -ftest-coverage)
     if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
-      target_link_options(mysofa-shared INTERFACE gcov --coverage)
+      target_link_options(mysofa-shared INTERFACE --coverage)
     else()
       target_link_libraries(mysofa-shared LINK_PUBLIC gcov --coverage)
     endif()
```

If this change is acceptable I can open a pull request.

System information:
- OS: Arch Linux x86_64
- libmysofa: git master (currently at commit 6e4381b)
- Compiler: gcc 9.2.1
- CMake: 3.16.4

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -89,7 +89,7 @@ if(UNIX
   target_compile_options(mysofa-static PUBLIC -g -O0 -Wall -fprofile-arcs
                                               -ftest-coverage)
   if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
-    target_link_options(mysofa-static INTERFACE gcov --coverage)
+    target_link_options(mysofa-static INTERFACE --coverage)
   else()
     target_link_libraries(mysofa-static LINK_PUBLIC gcov --coverage)
   endif()
@@ -118,7 +118,7 @@ if(BUILD_SHARED_LIBS)
     target_compile_options(mysofa-shared PUBLIC -g -O0 -Wall -fprofile-arcs
                                                 -ftest-coverage)
     if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
-      target_link_options(mysofa-shared INTERFACE gcov --coverage)
+      target_link_options(mysofa-shared INTERFACE --coverage)
     else()
       target_link_libraries(mysofa-shared LINK_PUBLIC gcov --coverage)
     endif()

